net/xia/ppal_ether/main.c

```diff
@@ -76,6 +76,95 @@ static int ether_vxt __read_mostly = -1;
 /* ETHER_FIB table internal operations. */
 static const struct xia_ppal_rt_iops *ether_rt_iops = &xia_ppal_list_rt_iops;
 
+/* Local-FXID. */
+struct fib_xid_ether_local {
+	struct xip_dst_anchor   xel_anchor;
+
+	/* WARNING: @xel_common is of variable size, and
+	 * MUST be the last member of the struct.
+	 */
+	struct fib_xid          xel_common;
+};
+
+static inline struct fib_xid_ether_local *fxid_lether(struct fib_xid *fxid)
+{
+	return likely(fxid)
+		? container_of(fxid, struct fib_xid_ether_local, xel_common)
+		: NULL;
+}
+
+/* ETHER local table operations. */
+static int local_newroute(struct xip_ppal_ctx *ctx,
+			  struct fib_xid_table *xtbl,
+			  struct xia_fib_config *cfg)
+{
+	struct fib_xid_ether_local *leid;
+	int rc;
+
+	leid = ether_rt_iops->fxid_ppal_alloc(sizeof(*leid), GFP_KERNEL);
+	if (!leid)
+		return -ENOMEM;
+	fxid_init(xtbl, &leid->xel_common, cfg->xfc_dst->xid_id,
+		  XRTABLE_LOCAL_INDEX, 0);
+	xdst_init_anchor(&leid->xel_anchor);
+
+	rc = ether_rt_iops->fib_newroute(&leid->xel_common, xtbl, cfg, NULL);
+	if (rc)
+		fxid_free_norcu(xtbl, &leid->xel_common);
+
+	return rc;
+}
+
+static void local_free_ether(struct fib_xid_table *xtbl, struct fib_xid *fxid)
```

Comment:
1. Could you add the following comment just before `local_free_ether()`:
`/* Don't call this function! Use free_fxid instead. */`

2. Place `local_free_ether()` just after `local_dump_ether()` as other principals do.